src/components/Shared/Form/Form.js

```diff
@@ -2,28 +2,44 @@ import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import Input from '../Input/Input';
 import { useDispatch } from 'react-redux';
+import loginActions from '../../../store/login/action';
+import registerActions from '../../../store/register/action';
 
 function Form(props) {
-  const { title, inputArr, tertiaryButton, actions } = props;
+  const { title, inputArray, tertiaryButton } = props;
   const [isValid, setIsValid] = useState('');
   const [amountTrueArray, setAmountTrueArray] = useState([]);
 
   const dispatch = useDispatch();
 
   useEffect(() => {
-    dispatch(actions.logout());
+    if (title === 'Login') {
+      dispatch(loginActions.logout());
+    }
   }, [dispatch]);
 
   const onSubmit = (event) => {
-    const inputLength = inputArr.length;
+    event.preventDefault();
+    const inputLength = inputArray.length;
     const trueLength = amountTrueArray.length;
 
     if (inputLength === trueLength) {
-      amountTrueArray.map((inputObject) => {
-        const { field } = inputObject;
-      });
       setIsValid(true);
-      dispatch(actions.login());
+
+      if (title === 'Login') {
+        dispatch(
+          loginActions.login(amountTrueArray[0].value, amountTrueArray[1].value)
+        );
+      } else {
+        dispatch(
+          registerActions.register(
+            amountTrueArray[0].value,
```

Comment:
Let's do some pair programming tomorrow to rework this components a bit. amountTrueArray is a bit confusing to read